siva/location.go

```diff
@@ -243,10 +205,31 @@ func (l *Location) repository(
 	id borges.RepositoryID,
 	mode borges.Mode,
 ) (borges.Repository, error) {
-	fs, err := l.getRepoFs(id, mode)
+	fs, err := l.getRepoFS(id, mode)
 	if err != nil {
 		return nil, err
 	}
 
 	return NewRepository(id, fs, mode, l)
 }
+
+func (l *Location) getRepoFS(id borges.RepositoryID, mode borges.Mode) (sivafs.SivaFS, error) {
+	if !l.lib.transactional || mode != borges.RWMode {
```

Comment:
@jfontan For only read operations, `FS()` could apply the checkpoint concurrently so I'm not sure if we should make checkpoint thread safe or not

Reply:
We must make sure that checkpoint actions are only called by one repo. I believe we can do that adding mutex to location so `cacheFS` invalidation and creation are safe and making sure that already closed repos cannot call `Commit`/`Close` again. Repository also needs a mutex for this.

Reply:
The repository `Commit/Close` is already added.

About the location mutex, I'll add it to be used only in the `FS()` method since it's the only point of conflict. `cacheFS`  is also set to nil in `Commit` and `Rollback` methods but it doesn't represent a concurrency problem becuse it's done at the end of the transaction to force refreshing the cache.

Reply:
Can you add some tests for these cases? Like double close.